### PR TITLE
fix: move workflow link to source query tab

### PR DESCRIPTION
**Pull request by zhangvi7**
### [ER-7007] Minor Querybook Improvements for Tier 1 tables
-  Currently all Tier 1 tables have a link to the workflow creating the table. However this information is not included under the Source Query tab

#### Test Plan

Added workflow link to bottom of source query tab

Screenshot 2024-05-01 at 11 25 32 PM

### CODE REVIEW

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx [hunk 1]
```diff
@@ -80,6 +81,21 @@ export const DataTableViewSourceQuery: React.FunctionComponent<IProps> = ({
         );
     });
 
+    const customProperties = table.custom_properties ?? {};
+    const workflowValue = customProperties['workflow']?.toString() ?? null;
```
Comment (jczhong84) [inline querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx:85]: nit: the workflow is already a string I think?

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx [hunk 2]
```diff
@@ -100,6 +116,7 @@ export const DataTableViewSourceQuery: React.FunctionComponent<IProps> = ({
     return (
         <div className="DataTableViewSourceQuery">
             {parentDOM}
+            {workflowDOM}
```
Comment (jczhong84) [inline querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx:119]: maybe better to put above the data job and source query?

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx [hunk 3]
```diff
@@ -80,6 +81,21 @@ export const DataTableViewSourceQuery: React.FunctionComponent<IProps> = ({
         );
     });
 
+    const customProperties = table.custom_properties ?? {};
+    const workflowValue = customProperties['workflow']?.toString() ?? null;
+    const workflowDOM = workflowValue ? (
+        <div className="DataTableViewSourceQuery-workflow">
+            <div className="data-job-source-query-top">
```
Comment (jczhong84) [inline querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx:88]: this class `data-job-source-query-top` is from component `DataJobMetadataInfo`, looks weird to be here. better to define its own.

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx (hunk 1 above)
Comment (zhangvi7) [reply querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx:85]: custom_property values are string | number, so change to string type for Link component

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx [hunk 4]
```diff
@@ -80,6 +81,21 @@ export const DataTableViewSourceQuery: React.FunctionComponent<IProps> = ({
...
+    const customProperties = table.custom_properties ?? {};
+    const workflowValue = customProperties['workflow']?.toString() ?? null;
+    const workflowDOM = workflowValue ? (
+        <div className="DataTableViewSourceQuery-workflow">
+            <div className="data-table-workflow">
+                <Title size="med" className="mb12">
+                    Workflow
+                </Title>
+            </div>
+            <Link to={workflowValue} newTab>
```
Comment (czgu) [inline querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx:93]: check if this is a link before making it into a link

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx (hunk 1 above)
Comment (czgu) [reply querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx:85]: @zhangvi7 it doesn't make sense to convert if it is a number

Comment (zhangvi7) [reply querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx]: updated to use typeof to assert string type

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx [hunk 5]
```diff
@@ -80,6 +81,26 @@ export const DataTableViewSourceQuery: React.FunctionComponent<IProps> = ({
...
+    const workflowValue = customProperties['workflow'];
+    const workflowDOM =
+        workflowValue && typeof workflowValue === 'string' ? (
+            <div className="DataTableViewSourceQuery-workflow">
+                <div className="data-table-workflow">
+                    <Title size="med" className="mb12">
+                        Workflow
+                    </Title>
+                </div>
+                {/https?:\/\/[^\s]+/.test(workflowValue.trim()) ? (
```
Comment (czgu) [inline querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx:94]: use isValidUrl in lib/utils/

**Review by czgu — COMMENTED** [review]
nice!

File: querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx (hunk 5 above)
Comment (zhangvi7) [reply querybook/webapp/components/DataTableViewSourceQuery/DataTableViewSourceQuery.tsx]: thanks! added

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**